bld/namelist_files/namelist_defaults_ctsm.xml

```diff
@@ -284,9 +289,14 @@ attributes from the config_cache.xml file (with keys converted to upper-case).
 <snowveg_flag phys="clm5_0" >ON_RAD</snowveg_flag>
 
 <!-- Snow pack settings-->
-<nlevsno phys="clm5_0" >12</nlevsno>
+<nlevsno phys="clm5_0" structure="standard">12</nlevsno>
+<nlevsno phys="clm5_0" structure="fast">5</nlevsno>
 <nlevsno phys="clm4_5" >5</nlevsno>
-<h2osno_max phys="clm5_0" >10000.0</h2osno_max>
+<!-- h2osno_max is more 'configuration' than 'structure'. But since it's
+     tied to nlevsno, we're controlling it via 'structure', like
+     nlevsno, so that defaults for the two remain consistent. -->
+<h2osno_max phys="clm5_0" structure="standard">10000.0</h2osno_max>
+<h2osno_max phys="clm5_0" structure="fast">5000.0</h2osno_max>
```

Comment:
This is maybe a wider question a nit picky one. But, something I do try to do is to keep important columns to line up. So in this case I like to have the numbers line up by adding extra spaces...

``` xml
<h2osno_max phys="clm5_0" structure="standard">10000.0</h2osno_max>
<h2osno_max phys="clm5_0" structure="fast" .       >5000.0</h2osno_max>
```

To me this makes it easier to read the numbers. But, not everyone agrees with this and for example pylint actually calls this kind of lining up by column noncompliance with the standard.

So I mainly bring it up for discussion purposes. I'd like to decide on our views on this. We could try to always do it, never do it, leave it as is, or let individuals do it differently.

Reply:
Done here and elsewhere in my changes in this file.

In this case I agree with you.

As a general rule, I absolutely agree that doing this enhances readability. It carries some cost to add, though for the typical things we align in the Fortran code, I have macros that let me highlight a block and run a single command to vertically align everything properly. If I didn't have those macros, I would argue against it, and I am hesitant to set this as a coding standard unless we provide macros like this for all common editors.

On the other hand, I find that it has a pretty high maintenance cost, for two reasons:

(1) It takes more time to add code or move code around when you need to vertically align everything around it. A very common case is adding a new variable declaration, associate line, etc., with a variable name that is longer than anything else currently there. Then you have to realign all the lines around it.

(2) Even more significantly, this realignment of existing code is a common cause of merge conflicts, which are the bane of my existence. True, these whitespace conflicts are easy enough to resolve once you figure out what happened, but I find it can waste significant time to figure out what happened and be confident in the merge resolution nonetheless.

All in all, I support vertical alignment as a general guideline but not as a hard-and-fast rule. I tend to vertically align most new code that I add, but often don't realign existing blocks of code when I add a new variable, mostly for reason (2): the merge cost does not seem worth the benefit in that case.

A quick google search turned up similar feelings: https://stackoverflow.com/questions/101958/code-formatting-is-lining-up-similar-lines-ok and https://softwareengineering.stackexchange.com/questions/30029/vertical-alignment-yea-or-nay

Reply:
Good discussion. We should talk more Monday...